unlock-app/src/components/interface/connect/EmailAccount.tsx

```diff
@@ -124,6 +131,33 @@ const SignIn = ({
   shoudOpenConnectModal = false,
   checkoutService,
 }: SignInProps) => {
+  const [isEmailCodeStatus, setEmailCodeStatus] = useState('none')
+
+  const router = useRouter()
+
+  const context = useSelector(checkoutService, (state) => state?.context)
+
+  const url = new URL(
+    `${window.location.protocol}//${window.location.host}${router.asPath}`
+  )
+  const params = new URLSearchParams(url.search)
+  params.append(
+    'shouldOpenConnectModal',
+    encodeURIComponent(shoudOpenConnectModal)
+  )
+  if (context?.lock) {
+    params.set('lock', encodeURIComponent(context.lock.address))
+  }
+  url.search = params.toString()
+
+  const callbackUrl = url.toString()
```

Comment:
Since we already have the logic in the Google component please isolate and re-use.